base/commands/migration/migration_stages.go

```diff
@@ -233,6 +241,46 @@ func fetchMigrationStatus(ctx context.Context, ci *hazelcast.ClientInternal, mig
 	return strings.TrimSuffix(strings.TrimPrefix(string(r.(serialization.JSON)), `"`), `"`), nil
 }
 
+func fetchOverallProgress(ctx context.Context, ci *hazelcast.ClientInternal, migrationID string) (time.Duration, float32, error) {
+	q := fmt.Sprintf(`SELECT JSON_QUERY(this, '$.remainingTime'), JSON_QUERY(this, '$.completionPercentage') FROM %s WHERE __key='%s'`, StatusMapName, migrationID)
+	res, err := ci.Client().SQL().Execute(ctx, q)
+	if err != nil {
+		return 0, 0, err
+	}
+	it, err := res.Iterator()
+	if err != nil {
+		return 0, 0, err
+	}
+	if it.HasNext() {
+		// single iteration is enough that we are reading single result for a single migration
+		row, err := it.Next()
+		if err != nil {
+			return 0, 0, err
+		}
```

Comment:
I thought you've introduced a function to query SQL and return a single result

Reply:
You are referring to querySingleRow that only uses `row.Get(0)` to fetch single result. But in here I have two values to read that I could not reuse it.

Reply:
Maybe `querySingleRow` could be changed to return the row, not the first column.

Reply:
The other functions use that function in common, so it is re-used a couple of times. Shall we keep it as is?

Reply:
If it's not too hard to make that change, I would change function I've described and change the code where it is called.

Reply:
I didn't mean to say it is hard, I think it is not very useful.

Reply:
There are only 4 places `querySingleRow` is used, so not much code will be changed with that refactoring, and it will be more generic, so it can also be used ^.
You can also change its return value to return `[]any` to return all column values.

Reply:
https://github.com/hazelcast/hazelcast-commandline-client/pull/423/commits/c53f607d67077525ef6a3b9e419dd990fc635f14